drivers/eeprom/CMakeLists.txt

```diff
@@ -26,3 +26,8 @@ zephyr_library_sources_ifdef(CONFIG_EEPROM_FAKE eeprom_fake.c)
 zephyr_library_sources_ifdef(CONFIG_EEPROM_AT2X_EMUL eeprom_at2x_emul.c)
 
 zephyr_library_sources_ifdef(CONFIG_EEPROM_MB85RCXX eeprom_mb85rcxx.c)
+
+if(CONFIG_EEPROM_RV3028)
+  zephyr_library_sources(eeprom_rv3028.c)
+  zephyr_library_include_directories(${ZEPHYR_BASE}/drivers/rtc)
```

Comment:
Sorry, just saw this now. This is not the best way to do it, I'm afraid. It's true that we have some legacy implementations in tree doing something similar, but for this new functionality I'd suggest implementing a base MFD device driver, layering both the RTC and EEPROM drivers on top of that. This avoids these somewhat weird dependencies between an EEPROM driver and a RTC driver.

Reply:
so `drivers/mfd/mfd_rv3028.c` and

```
if(CONFIG_EEPROM_RV3028)
  zephyr_library_sources(eeprom_rv3028.c)
  zephyr_library_include_directories(${ZEPHYR_BASE}/drivers/mfd)
  ...
```

Reply:
No. The header file goes under `include/zephyr/drivers/mfd/`.

Reply:
Private how? More private than the other MFD driver APIs in the three, which are placed in the include folder?

Specifying the MFD driver API in a public header file allows for easier out-of-tree usage and prevents odd dependencies between otherwise unrelated driver subsystems.

Reply:
I do agree, but was this not discussed earlier https://github.com/zephyrproject-rtos/zephyr/pull/74034#discussion_r1639557003 where it seemed we wanted to keep the header private? @nordicjm?

Reply:
ok, in that case, the `drivers/rtc/rtc_rv3028.h` header does seem fine if the names are updated to use the `mfd_` namespace, and can be moved to `zephyr/drivers/mfd` I believe, maybe with the name of the lock/unlock functions updated to not include the `_sem_` part? locking is not a private implementation detail IMO :)

There is precedent for something a bit wilder in there:
https://github.com/zephyrproject-rtos/zephyr/blob/318b49570a77ce631dc3d97ad7e7a05df5459d8e/include/zephyr/drivers/mfd/nct38xx.h#L25
and
https://github.com/zephyrproject-rtos/zephyr/blob/318b49570a77ce631dc3d97ad7e7a05df5459d8e/include/zephyr/drivers/mfd/adp5585.h#L102-L104

we may want to start setting and documenting some rules for mfd drivers :)

Reply:
Sounds good to me.